lib/web/mage/backend/tabs.js

```diff
@@ -65,7 +65,7 @@ define([
                 this.anchors :
                 this._getList().find('> li > a[href]');
 
-            return anchors.index($('#' + id));
+            return anchors.index($('a#' + id));
```

Comment:
Identifiers must be unique on each page. Adding a qualifier to something that's already unique doesn't seem to make sense to me. Perhaps the fix for the problem that you're experiencing is to remove elements or change identifiers to resolve the duplicate problem.

Reply:
I'm not sure I'm following when you say "duplicate". I am just trying to grab the element with the correct identifier that the find returns. 

![image](https://user-images.githubusercontent.com/13631233/181600554-c841c46b-b446-4dc8-a867-b47ea259fe95.png)

For example above on a product attribute edit page, looking at the anchors that get found, they're all in the form of `a#product_attribute_tabs_X`
Looking at the `id` variable, the `index` method doesn't find anything and always returns a `-1`, because it can't find `#product_attribute_tabs_X`, as the `a` is in front of it.

This makes all edit pages that use the `tabs.js` script to never be able to link directly to a tab.

Reply:
`$` here is jQuery. Calling `jQuery()` as a function like this takes a CSS selector as its first argument. The string that's being generated / used here is in the form `#identifier`, which is an ID selector as described here: https://developer.mozilla.org/en-US/docs/Web/CSS/ID_selectors

See here for information about the `id` attribute. https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id (This page is one of many that states that IDs must be unique within a document / page.)

It sounds like this bug requires further investigation.

Reply:
I see what you're saying. The `li` and child `a` elements have the same identifier, making the index not find it.
That identifier defined in the backend extension in
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Backend/view/adminhtml/templates/widget/tabs.phtml#L34
and line 37 for the `a`.

So in reality, probably the actual solution would be to give the `a` tag a different identifier, and look for that new identifier when trying to find the index in the `tabs.js` widget.

Reply:
Yes, that sounds like the right thing to do. Or change (remove?) the `id` attribute for the list item (`<li>`) and leave the anchor (`<a>`) with its current ID.

Reply:
Tabs fail to load if the id is different or missing on the `<li>` element, it looks like.

Reply:
> Yes, that sounds like the right thing to do. Or change (remove?) the `id` attribute for the list item (`<li>`) and leave the anchor (`<a>`) with its current ID.

```
class="admin__page-nav-item no-display"
                    id="<?= $block->escapeHtmlAttr($block->getTabId($_tab)) ?>"
```
=>

```
class="admin__page-nav-item no-display"
                    id="<?= $block->escapeHtmlAttr($block->getTabId($_tab)) ?>_nav_item"

```
```
$('li.admin__page-nav-item#{$block->escapeJs($block->getTabId($_tab))}')
```
=>
```
$('li.admin__page-nav-item#{$block->escapeJs($block->getTabId($_tab))}_nav_item')
```